license.go

```diff
@@ -15,31 +22,64 @@ func IsEnterpriseLicensedOrDevelopment(config *model.Config, license *model.Lice
 	return IsConfiguredForDevelopment(config)
 }
 
-// IsE10LicensedOrDevelopment returns true when the server is licensed with a Mattermost
-// Enterprise E10 License, or has `EnableDeveloper` and `EnableTesting` configuration settings
-// enabled, signaling a non-production, developer mode.
+// isValidSkuShortName returns whether the SKU short name is one of the known strings;
+// namely: E10 or professional, or E20 or enterprise
+func isValidSkuShortName(license *model.License) bool {
+	if license == nil {
+		return false
+	}
+
+	switch license.SkuShortName {
+	case e10, e20, professional, enterprise:
+		return true
+	default:
+		return false
+	}
+}
+
+// IsE10LicensedOrDevelopment returns true when the server is at least licensed with a legacy Mattermost
+// Enterprise E10 License or a Mattermost Professional License, or has `EnableDeveloper` and
+// `EnableTesting` configuration settings enabled, signaling a non-production, developer mode.
 func IsE10LicensedOrDevelopment(config *model.Config, license *model.License) bool {
 	if license != nil &&
-		license.Features != nil &&
-		license.Features.LDAP != nil &&
-		*license.Features.LDAP {
+		(license.SkuShortName == e10 || license.SkuShortName == professional ||
+			license.SkuShortName == e20 || license.SkuShortName == enterprise) {
 		return true
 	}
 
+	if !isValidSkuShortName(license) {
+		// As a fallback for licenses whose SKU short name is unknown, make a best effort to try
+		// and use the presence of a known E10/Professional feature as a check to determine licensing.
+		if license != nil &&
+			license.Features != nil &&
+			license.Features.LDAP != nil &&
+			*license.Features.LDAP {
+			return true
+		}
+	}
+
 	return IsConfiguredForDevelopment(config)
```

Comment:
If we moved this check to the top, I wonder if this would help with the flow? (0/5)